scheduling-manager/src/main/java/com/amazonaws/blox/scheduling/scheduler/engine/daemon/DaemonEnvironment.java

```diff
@@ -48,6 +54,30 @@ public StartTask startTaskFor(ContainerInstance i) {
         .build();
   }
 
+  public List<StopTask> stopTaskFor(final List<Task> tasks) {
+    final List<String> tasksToStop =
+        tasks
+            .stream()
+            .filter(this::unmatchesTask)
+            .map(Task::getTaskDefinitionArn)
+            .collect(Collectors.toList());
+    return tasksToStop
+        .stream()
+        .map(
+            t ->
+                StopTask.builder()
+                    .clusterName(environment.getClusterName())
+                    .task(t)
+                    .reason("replaceAfterTerminate")
+                    .build())
+        .collect(Collectors.toList());
+  }
+
+  public boolean unmatchesTask(Task t) {
+    return !t.getGroup().equals(environment.getEnvironmentName())
```

Comment:
We should only compare the task which is under the same environment.  If task is in different env name, they are managed by different environment and different config.   Should it be:
` return t.getGroup().equals(environment.getEnvironmentName()) && !t.getTaskDefinitionArn().equals(environment.getTaskDefinitionArn());`

Reply:
Sure. Good catch!